plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/actions/ExecuteSelectedCommandAction.java

```diff
@@ -57,8 +58,12 @@ public void updateInPerspective(ActionEvent event) {
     @Override
     public void actionPerformed(ActionEvent e) {
         final CommandConfiguration command = selectCommandAction.getSelectedCommand();
-        if (command != null) {
-            commandManager.execute(command);
+        MachineDto machine = selectCommandAction.getSelectedMachine();
+
+        if (command == null || machine == null) {
```

Comment:
https://github.com/google/guava/blob/master/guava/src/com/google/common/base/Preconditions.java#L107 better way to check arguments